x-pack/legacy/plugins/ml/public/services/job_messages_service.js

```diff
@@ -13,6 +13,29 @@
 import { ML_NOTIFICATION_INDEX_PATTERN } from 'plugins/ml/../common/constants/index_patterns';
 import { ml } from 'plugins/ml/services/ml_api_service';
 
+// filter to match job_type: 'anomaly_detector' or no job_type field at all
+// if no job_type field exist, we can assume the message is for an anomaly detector job
+const anomalyDetectorTypeFilter = {
+  bool: {
+    should: [
+      {
+        term: {
+          job_type: 'anomaly_detector'
+        }
+      },
+      {
+        bool: {
+          must_not: {
+            exists: {
+              field: 'job_type'
+            }
+          }
+        }
+      }
+    ]
+  }
```

Comment:
This might need the addition of `"minimum_should_match":1` I think.

Reply:
added in 65b03722df06c8929b7f3c2efe70cc16bf256dc9